### DISCUSSION TITLE
Can Rego policy, when using ignore_policy flag,  ignore vulnerability ID's ?

### DISCUSSION THREAD

**Discussion by akcrisp** [aquasecurity/trivy#6813] [Q&A] [answered]
### Question

I want to use the ignore_policy option to ignore particular cve's. Having read the docs and looked at the examples, (I note the link to the helper file is broken and module.go it references appears to be missing), its not clear to me whether you can simply have a rego file with an option something like this -

package trivy

import data.lib.trivy

default ignore  = false

ignore_vulnerabilities :=  {"CVE-1234-5678", "CVE-5678-1234","CVE-2345-6789"}

---

I want to do this using rego rather than the .trivyignore file as I am planning on add this file as an artifact to an image. Then I want to see if i can work with our cloud vendor to use the rego policy when doing an image scan to ignore those cve's. 

Can someone confirm if ignoring the cve (which seems to be the most simplest form of being able to ignore a vulnerability beyond all the cvss stuff it does support, does indeed work ??

Andy

### Target

Container Image

### Scanner

Vulnerability

### Output Format

None

### Mode

Standalone

### Operating System

Linux - various

### Version

```bash
0.48.2
```

**Comment by nikpivkin** [comment, marked as answer]
Hi @akcrisp !

Have you looked at the latest version of the documentation? https://aquasecurity.github.io/trivy/v0.51/docs/configuration/filtering/#by-rego

**Comment by akcrisp** [comment]
ah thanks for that - no i had missed it - so on that basis i can do

ignore input.VulnerabilityID[_] == 'CVE-2015-5186' 

Will give it ago.

**Reply by nikpivkin** [reply]
`VulnerabilityID` is a string, not an array, so there's that:
```rego
ignore {
    input.VulnerabilityID == "..."
}
```

**Reply by akcrisp** [reply]
good spot ! thanks

**Reply by akcrisp** [reply]
@nikpivkin  sorry quick follow up  - if i have more than 1 cve i want to ignore, do i need to repeat the whole ignore statement each time or is it possible to do somethig like - 

ignore {
       input.VulnerabilityID == "..."
       input.VulnerabilityID == "..."
}

I've tried it but although it doesn't error, it equally doesn't ignore the cve's, it does if only one cve entry or if i repeat the whole ignore section per cve ie

ignore {
       input.VulnerabilityID == "..."
}
ignore {
       input.VulnerabilityID == "..."
}

**Reply by nikpivkin** [reply]
The second option, so OR is presented through multiple rules with the same head. [This](https://www.styra.com/blog/how-to-express-or-in-rego/) will help you understand how rules work in Rego.

**Reply by akcrisp** [reply]
Actually @nikpivkin  I figured out how to do a list in opa (crash dive  in opa rules writing :-) ) - put this here in case anyone else finds it useful. 

package trivy

import data.lib.trivy

default ignore = false

cve_list := ["cve-xxxx-yyyy", "cve-xxxx-yyyy", "cve-xxxx-yyyy"]

ignore {
        input.VulberabilityID == cve_list[_]
}

**Reply by nikpivkin** [reply]
@akcrisp Or you can use the Rego v1 syntax.

```rego
package trivy

import rego.v1

cve_list := {"cve-xxxx-yyyy", "cve-xxxx-yyyy", "cve-xxxx-yyyy"}

default ignore := false

ignore if input.VulberabilityID in cve_list
```

**Reply by akcrisp** [reply]
@nikpivkin is there any advantage of using the rego v1 syntax ? is that the future syntax  ??

**Reply by nikpivkin** [reply]
You can find answers in the documentation https://www.openpolicyagent.org/docs/latest/opa-1/
